### DISCUSSION TITLE
DG: Error: write EPIPE - Node SDK

### DISCUSSION THREAD

**Discussion by jakejenn21** [deepgram/community#144] [General help] [answered]
I'm running a script in which I reference audio files in a local directory, create a buffer using `fs.readFileSync` and pass it to Deepgram's `transcription.preRecorded` function. I then get an error as a response. `DG: Error: write EPIPE`. In the github repo the issue I believe is being caught @ [Code](https://github.com/deepgram/deepgram-node-sdk/blob/main/src/httpRequest.ts#L117)  I'm unsure if this has to do with the file being around 20 MB (DG's docs says there is no limit) or I'm not preparing the audio file correctly to what the DG req is expecting. Has anyone encountered this problem?

Here is my code:

```
                        const buffer = fs.readFileSync('./' + pathToAudio);
			console.log(buffer);
			const audioSource = { buffer: buffer, mimetype: 'audio/m4a' };
			const dgResponse = await deepgram.transcription.preRecorded(
				audioSource,
				{
					tier: 'nova',
					punctuate: true,
					paragraphs: true,
					utterances: true,
					smart_format: true,
					detect_entities: true,
					numbers_spaces: false,
					measurements: true,
					date_format: '%Y-%m-%d',
					keywords: keywords,

				},
			);

			console.log(dgResponse);
```
![Screen Shot 2023-05-09 at 1 51 29 PM](https://github.com/deepgram/community/assets/47189949/db24f59d-0ec1-417d-808a-4828e50bf728)

**Comment by lukeocodes** [comment]
Try using this code, which is based on the code from [our guide](https://developers.deepgram.com/sdks-tools/sdks/node-sdk/pre-recorded-transcription/#sending-a-buffer)

```js
// Sending a Buffer of the file
fs.readFile('./' + pathToAudio, function (err, buffer) {
    const audioSource = { buffer: buffer, mimetype:'audio/m4a' };
});
```

i'm not sure why it would be different using `readFileSync`, though

**Comment by briancbarrow** [comment]
@jakejenn21 I just tried your code using a file that is ~53MB and it worked okay so I'm not sure what is happening here. What environment are you running in? What version of Node are you on and what version of the Deepgram SDK are you using? 

When you log into [your Deepgram console](https://console.deepgram.com/) do you see those calls making it through?

**Comment by jakejenn21** [comment, marked as answer]
@briancbarrow  @lukeocodes Thanks for your comments, I found this error after commenting out the options. My keywords array had some inconsistencies. More descriptive error handling + stack trace would've helped.

**Reply by lukeocodes** [reply]
Thanks Jake, appreciate the feedback. We'll look at making the error response and trace more helpful.
